pyomo/core/plugins/transform/scaling.py

```diff
@@ -81,16 +81,61 @@ def _create_using(self, original_model, **kwds):
         self._apply_to(scaled_model, **kwds)
         return scaled_model
 
+    def _suffix_finder(self, component_data, suffix_name, top_level=None):
+        """
+        Find suffix value for a given component data object in model tree according
+        to the following precedence:
+
+        1. Specific suffix values at higher levels take precedence.
+        2. If no specific suffix value is found, the lowest level default value
+        (Suffix key None) is used.
+        3. If no suffix values are defined, return None.
+
+        Args:
+            component_data: component data object to find suffix value for.
+            suffix_name: name of Suffix to search for.
+            top_level: Block at which to stop searching for Suffixes (default=None).
+
+        Returns:
+            Value for suffix associated with component data if found, else None.
+        """
+        # Prototype for Suffix finder
+        val = None  # default return of None if no corresponding suffix found
+
+        # Walk parent tree and search for suffixes
+        parent = component_data.parent_block()
+        # TODO: Stop at instance, or run out the top of the stack?
+        # For now, run out the top
+        while parent is not top_level:
```

Comment:
Note that this will NOT look for suffixes at the top_level if top_level is not None...